features/netsocket/SocketAddress.cpp

```diff
@@ -215,17 +215,19 @@ void SocketAddress::set_port(uint16_t port)
 
 const char *SocketAddress::get_ip_address() const
 {
-    char *ip_address = (char *)_ip_address;
+    if (_addr.version == NSAPI_UNSPEC) {
```

Comment:
So UNSPEC SocketAddress round-trips via NULL string pointer, if I read correctly. Seems fine. But would still vaguely prefer a real string, so users don't have to take care when doing `printf("%s", foo->get_ip_address())`.